packages/fetchai/skills/confirmation_aw2/registration_db.py

```diff
@@ -65,26 +65,28 @@ def _initialise_backend(self) -> None:
 
     def set_trade(
         self, address: str, timestamp: datetime.datetime, data: Dict[str, str],
-    ):
+    ) -> None:
         """Record a registration."""
         record = self.get_trade_table(address)
         if record is None:
-            return
-
-        _, first_trade, second_trade, first_info, second_info = record
-        is_second = first_trade is not None and second_trade is None
-        if is_second:
+            command = "INSERT INTO trade_table(address, first_trade, second_trade, first_info, second_info) values(?, ?, ?, ?, ?)"
+            variables: Tuple[
+                str, datetime.datetime, Optional[datetime.datetime], str, Optional[str]
+            ] = (address, timestamp, None, json.dumps(data), None)
+        else:
+            _, first_trade, second_trade, first_info, _ = record
+            is_second = first_trade is not None and second_trade is None
+            is_more_than_two = first_trade is not None and second_trade is not None
+            if is_more_than_two or not is_second:
+                return
             command = "INSERT INTO trade_table(address, first_trade, second_trade, first_info, second_info) values(?, ?, ?, ?, ?)"
```

Comment:
@DavidMinarsch this should be `INSERT or REPLACE into ...`

Reply:
notice also that `address` now is a primary key

Reply:
Yes!